src/once_cell.rs

```diff
@@ -0,0 +1,92 @@
+use crate::Python;
+use std::cell::UnsafeCell;
+
+/// A write-once cell similar to `once_cell::OnceCell`.
+///
+/// Unlike `once_cell::sync` which uses locks to achieve thread safety, this implementation uses
+/// the Python GIL to mediate concurrent access.
```

Comment:
Let's make it clear why we need GIL(i.e., the interaction between GIL and another mutex can cause a deadlock).
Or it is also good to link the FAQ section in the guide.